kajas/src/app/registration/registration.component.ts

```diff
@@ -0,0 +1,210 @@
+import { Component, OnInit, OnDestroy, Renderer2 } from '@angular/core';
+import { FormBuilder, FormGroup, Validators, AbstractControl, ValidationErrors } from '@angular/forms';
+import { ModalService } from '../../services/modal.service';
+import { UserService } from '../../services/user.service';
+import axios from 'axios';
+
+@Component({
+  selector: 'app-registration',
+  templateUrl: './registration.component.html',
+  styleUrls: ['./registration.component.css']
+})
+export class RegistrationComponent implements OnInit, OnDestroy {
+  registrationForm: FormGroup;
+  hidePassword: boolean = true;
+  hideConfirmPassword: boolean = true;
+  existingUsernames: string[] = [];
+  showModal: boolean = false;
+  modalMessage: string = '';
+
+  constructor(
+    private fb: FormBuilder,
+    private renderer: Renderer2,
+    private modalService: ModalService,
+    private userService: UserService 
+  ) {
+    this.registrationForm = this.fb.group({
+      username: ['', [Validators.required, this.usernameValidator.bind(this)]],
+      email: ['', [Validators.required, Validators.email, this.gmailValidator]],
+      password: ['', [Validators.required, this.passwordValidator]],
+      confirmPassword: ['', [Validators.required, this.confirmPasswordValidator.bind(this)]],
+      firstName: ['', Validators.required],
+      middleName: [''],
+      lastName: ['', Validators.required],
+      terms: [false, Validators.requiredTrue]
+    });
+  }
+
+  ngOnInit(): void {
+    this.setInitialStyles();
+    window.addEventListener('resize', this.applyBackground.bind(this));
```

Comment:
### Stop using Vanilla JS way of doing things.
- Get rid of using querySelectors and addEventsListener.
[https://docs.google.com/presentation/d/1Mdcq3WcHksBlbCvxi6wx5vajn4v8LGlJRnBIu3x4fvw/edit#slide=id.g2e205e48566_0_29](url)